ldk/javascript/src/whisper/react/renderer.test.tsx

```diff
@@ -0,0 +1,357 @@
+            body: 'markdown.body',
+          },
+          {
+            type: WhisperComponentType.Markdown,
+            body: 'nakedmarkdown.body',
+          },
+        ],
+      });
+      expect(onMount).toHaveBeenCalled();
+    });
+    it('ignores whispers rendered underneath whispers', async () => {
+      await render(
+        <oh-whisper label="whisper.label" onClose={() => {}}>
+          <oh-whisper label="forbiddenwhisper.label" onClose={() => {}}>
+            forbidden.text
+          </oh-whisper>
+          nakedmarkdown.body
+        </oh-whisper>,
+        whisperInterface,
+      );
+      expect(whisperInterface.createOrUpdateWhisper).toHaveBeenCalledWith({
+        label: 'whisper.label',
+        onClose: expect.any(Function),
+        components: [
+          {
+            type: 'markdown',
+            body: 'nakedmarkdown.body',
+          },
+        ],
+      });
+    });
+    it('generates a whisper with a class component correctly', async () => {
+      const onMount = jest.fn();
+      await render(
+        <oh-whisper label="whisper.label" onClose={() => {}}>
+          <ButtonClass label="button.label" onMount={onMount} onRender={jest.fn()} />
+          <oh-markdown body={'markdown.body'} />
+          nakedmarkdown.body
+        </oh-whisper>,
+        whisperInterface,
+      );
+
+      expect(whisperInterface.createOrUpdateWhisper).toHaveBeenCalledWith({
+        label: 'whisper.label',
+        onClose: expect.any(Function),
+        components: [
+          {
+            type: WhisperComponentType.Button,
+            label: 'button.label',
+            onClick: expect.any(Function),
+          },
+          {
+            type: WhisperComponentType.Markdown,
+            body: 'markdown.body',
+          },
+          {
+            type: WhisperComponentType.Markdown,
+            body: 'nakedmarkdown.body',
+          },
+        ],
+      });
+      expect(onMount).toHaveBeenCalled();
+    });
+  });
+  describe('calling events triggers state updates', () => {
+    it('generates a whisper with a functional component correctly', async () => {
+      const deferred = createDeferred();
+      const onMount = jest.fn(() => {
+        deferred.resolve(true);
+      });
+      const onRender = jest.fn(() => {});
+      await render(
+        <oh-whisper label="whisper.label" onClose={() => {}}>
+          <ButtonFunctional label="button.label" onMount={onMount} onRender={onRender} key="f" />
+          <oh-markdown key="markdown" body="markdown.body" />
+          nakedmarkdown.body
+        </oh-whisper>,
+        whisperInterface,
+      );
+      await deferred.promise;
+
+      const button = mocked(whisperInterface.createOrUpdateWhisper).mock.calls[0][0]
+        .components[0] as Button;
+      button.onClick(undefined, null as any);
+
+      expect(whisperInterface.createOrUpdateWhisper).toHaveBeenCalledWith({
+        label: 'whisper.label',
+        onClose: expect.any(Function),
```

Comment:
Could you check that the same `onClose` function is being sent to the whisperInterface? Or is the reference to that function being lost in mapping?

Reply:
Probably don't have to cover it in every test but it could be worthwhile instead of just `expect.any(Function)`

Reply:
The identity does change here, I intercept it in the WhisperInstanceWrapper